kratos/utilities/sensitivity_builder.cpp

```diff
@@ -291,9 +294,12 @@ void CalculateNonHistoricalSensitivities(const SensitivityVariables<TDataType>&
             return;
         builder.CalculateLocalSensitivity(*rVariables.pDesignVariable, rElement,
                                           rResponseFunction, rProcessInfo);
-        builder.LocalSensitivity *= ScalingFactor;
-        AssembleOnDataValueContainer(*rVariables.pOutputVariable,
-                                     builder.LocalSensitivity, rElement.Data());
+        // if rElement does not contribute to local sensitivity, skip assembly
+        if (builder.LocalSensitivity.size() != 0){
```

Comment:
This will not work, since in the case a element does not contribute to the sensitivity the size of the pseudo-load computed in `CalculateSensitivityMatrix` is not sized to zero. @dbaumgaertner I will contact you in person in order to talk about a concept on how to deal with this topic because its also on my TODO-list.